apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/EntityDbTable.java

```diff
@@ -216,8 +215,12 @@ private T get(Connection con, PreparedStatement pstmt, boolean cache) throws SQL
                     dataSource.getCache(table).put(dbKey, t);
                 }
             }
-            if (rs.next()) {
-                throw new RuntimeException("Multiple records found");
+            //TODO: check this
+            // OL: there are a lot of duplicates really in PUBLIC_KEY table, one with real key and next with null
+            // I do not now why old code does not react on this. So we just log, not thow here
+            if (rs.next() && dbKey!=null) {
```

Comment:
Its normal behavior for derived tables. These tables represents  history logs, so multiple values can be stored at same time. But only one value has flag  `latest=true` and this value used by application. When rollback occurred, some values with `latest=true` can be deleted, so old value with `latest=false` will be updated to `latest=true` and used by application.